File: JAVA/1962. Remove Stones to Minimize the Total.java
class Solution {
    public int minStoneSum(int[] p, int k) {
        PriorityQueue<Integer>x=new PriorityQueue<>(Collections.reverseOrder());
        for(int i:p){
            x.add(i);
        }
        for(int i=0;i<k;i++){
            int a=x.poll();
            a-=(int)Math.floor(a/2);
            x.add(a);
        }
        int s=0;
        while(x.size()!=0){
            s+=x.poll();
        }
        return s;
    }
}

File: C++/1962. Remove Stones to Minimize the Total.cpp
class Solution {
public:
    int minStoneSum(vector<int>& p, int k) {
        priority_queue<int>x(p.begin(),p.end());
        for(int i=0;i<k;i++){
           int a=x.top();
           x.pop();
           a-=a/2;
           x.push(a);
        }
        int s=0;
        while(!x.empty()){
            s+=x.top();
            x.pop();
        }
        return s;
    }
};